### ISSUE TITLE
 Extending git / git plugins / git commands

### ISSUE THREAD

**Issue by LoveIsGrief** [progit/progit2#723]
Hello,

This is a continuation of git/git-scm.com#744 

> A chapter on extending git e.g `git-my-command` would be nice. There doesn't seem to be a reference or guide on doing it. There's always [stack overflow](https://stackoverflow.com/questions/10978257/extending-git-functionality), but that seems to be just a pointer in the right direction.

Basically a new (sub) chapter in **8. Customizing git** with the above title (or similar) would be pretty nice to have. Or it could possibly be put in **10. Git Internals** since knowing the internals of git can be very helpful.

Would this be acceptable in the book?

**Comment by ben** [comment]
Yeah, that'd be a nice addition. I think Chapter 8 is the right place for it; you don't technically need to know plumbing commands to write an extension.

**Comment by LoveIsGrief** [comment]
Alright, I'll add it to my TODO list and see when I get around to doing it :)

**Comment by KZiemian** [comment]
@LoveIsGrief Should this issue remains open?

**Comment by LoveIsGrief** [comment]
@KZiemian unfortunately I never had time to pick this up. IMO, it can stay open as such documentation could be useful to others, but I won't have time to do it.

**Comment by KZiemian** [comment]
I will add it to my TODO list. It doesn't mean that I will write this text, but maybe by some miracle this would happen?

**Comment by mprimi** [comment]
Bumping this. As of today there is **still no mention of plugins/extensions at all** on the site. It would be really useful to have some.

Personally, I'm interested in instructions for installing plugins (probably simpler guide than _developing_ plugins).

I could not find much online. If someone has suggestions, I'd happily take them, until an official guide is available.

(I understand and appreciate this is volunteer work, thank you for even considering doing it @KZiemian)
